webgpu.c

```diff
@@ -1583,6 +1600,93 @@ wasi_webgpu_webgpu_gpu_feature_name_t featureNativeToWasi(WGPUFeatureName const
+    const char* texture_compression_etc2 = "texture-compression-etc2";
+    const char* texture_compression_astc = "texture-compression-astc";
+    const char* texture_compression_astc_sliced3d = "texture-compression-astc-sliced3d";
+    const char* timestamp_query = "timestamp-query";
+    const char* indirect_first_instance = "indirect-first-instance";
+    const char* shader_f16 = "shader-f16";
+    const char* rg11b10ufloat_renderable = "rg11b10ufloat-renderable";
+    const char* bgra8unorm_storage = "bgra8unorm-storage";
+    const char* float32_filterable = "float32-filterable";
+    const char* float32_blendable = "float32-blendable";
+    const char* clip_distances = "clip-distances";
+    const char* dual_source_blending = "dual-source-blending";
+    const char* subgroups = "subgroups";
+
+    const char* c_string = NULL;
+
+    switch (feature) {
+        case WGPUFeatureName_DepthClipControl:
+            c_string = depth_clip_control;
+            break;
+        case WGPUFeatureName_Depth32FloatStencil8:
+            c_string = depth32float_stencil8;
+            break;
+        case WGPUFeatureName_TextureCompressionBC:
+            c_string = texture_compression_bc;
+            break;
+        case WGPUFeatureName_TextureCompressionBCSliced3D:
+            c_string = texture_compression_bc_sliced3d;
+            break;
+        case WGPUFeatureName_TextureCompressionETC2:
+            c_string = texture_compression_etc2;
+            break;
+        case WGPUFeatureName_TextureCompressionASTC:
+            c_string = texture_compression_astc;
+            break;
+        case WGPUFeatureName_TextureCompressionASTCSliced3D:
+            c_string = texture_compression_astc_sliced3d;
+            break;
+        case WGPUFeatureName_TimestampQuery:
+            c_string = timestamp_query;
+            break;
+        case WGPUFeatureName_IndirectFirstInstance:
+            c_string = indirect_first_instance;
+            break;
+        case WGPUFeatureName_ShaderF16:
+            c_string = shader_f16;
+            break;
+        case WGPUFeatureName_RG11B10UfloatRenderable:
+            c_string = rg11b10ufloat_renderable;
+            break;
+        case WGPUFeatureName_BGRA8UnormStorage:
+            c_string = bgra8unorm_storage;
+            break;
+        case WGPUFeatureName_Float32Filterable:
+            c_string = float32_filterable;
+            break;
+        case WGPUFeatureName_Float32Blendable:
+            c_string = float32_blendable;
+            break;
+        case WGPUFeatureName_ClipDistances:
+            c_string = clip_distances;
+            break;
+        case WGPUFeatureName_DualSourceBlending:
+            c_string = dual_source_blending;
+            break;
+        case WGPUFeatureName_Subgroups:
+            c_string = subgroups;
+            break;
+        default:
+            unreachable();
+    }
+
+    imports_string_t output = {};
```

Comment:
```suggestion
    imports_string_t output = {};
    size_t len = strlen(c_string);
    output.ptr = malloc(len);
    memcpy(output.ptr, c_string, len);
    output.len = len;
    return output;
```
Note should cache len